stratum/hal/lib/barefoot/bf_sde_interface.h

```diff
@@ -177,6 +177,17 @@ class BfSdeInterface {
   // Disable a port.
   virtual ::util::Status DisablePort(int device, int port) = 0;
 
+  // Set the port shaping properties on a port.
+  // If in_pps is true, the burst size and rate is measured in packets and pps.
+  // Else, it's in bytes and kbps.
+  virtual ::util::Status SetPortShapingRate(int device, int port, bool in_pps,
+                                            uint32 burst_size,
+                                            uint32 rate_per_second) = 0;
```

Comment:
We might want to keep the rate in bps to be consistent with other uses throughout.